onebusaway-android/src/main/java/org/onebusaway/android/app/Application.java

```diff
@@ -623,4 +624,13 @@ private void createNotificationChannels() {
         }
     }
 
+    public static boolean isIgnoringBatteryOptimizations(Context applicationContext) {
+        PowerManager pm = (PowerManager) applicationContext.getSystemService(Context.POWER_SERVICE);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
+                pm.isIgnoringBatteryOptimizations(applicationContext.getPackageName())) {
+            return true;
+        }
+
+        return false;
```

Comment:
We should think about how to record versions of Android `< Build.VERSION_CODES.M`.  

Right now it returns `false`, but in reality battery optimizations didn't exist in their current form until Android O.  Our purpose of recording this is to understand if we might be missing device behavior because the user restricted OBA from running in the background (a `false` value on Android M and higher).  But, because we're not recording the version of Android in the data, we may end up with a lot of `false` values for legacy devices.  In that case, we won't know if it's because the value is coming from a legacy device, or if the user has restricted OBA on a newer device.

Options would be to return `true` for legacy devices (because they really don't have the battery optimizations), or record the version of Android along with the `isIgnoringBatteryOptimizations()` value.

Reply:
Per our discussion we'll allow `null` values for legacy devices.

Reply:
Just marking these as `done` for future reference.